ts/utils/messages.ts

```diff
@@ -146,3 +160,47 @@ export const getPrescriptionDataFromName = (
     return none;
   });
 };
+
+/**
+ * extract the CTAs if they are nested inside the message markdown content
+ * if some CTAs are been found, the localized version will be returned
+ * @param message
+ * @param locale
+ */
+export const getCTA = (
+  message: CreatedMessageWithContent,
+  locale: Locales = I18n.currentLocale()
+): Option<CTAS> => {
+  return fromPredicate((t: string) => FM.test(t))(message.content.markdown)
+    .map(m => FM<MessageCTA>(m).attributes)
+    .chain(attrs => fromNullable(attrs[locale]));
```

Comment:
In case we are working on a locale that is missing from the CTA we received in the message we return a `none` and we won't show the cta even if it has been added to the message.

E.g. User changed to "EN" from preferences option in the message we received just the CTA on IT language -> no CTA displayed. Should we fallback on the received CTA?

Reply:
@CrisTofani 
Yes it could happen.
What if we make it/en as required field?

Reply:
I think it could fail anyway cause we don't have the complete control of what we get since it is a remote content, maybe we should handle `attrs` as an array of string and fallback on the first we get?

Reply:
@CrisTofani 
starting from you warning I improved the logic and I added a test https://github.com/pagopa/io-app/pull/1936/commits/ae77f58dc12fa965722f15cb588029a8582870c0:
now if the front-matter doesn't respect MessageCTA type the decoding will fail.
So if, from remote, we receive a data that doesn't contain **it** nor **en** the CTA won't be decoded.

Btw about the fallback we can do as follow: since at the moment the MessageCTA requires it/en, if the CTA for the current locale is not available we can fallback to the other one. if from remote we receive not **it**/**en**, **fr** for example, the decoding will fail

What do you think?

Reply:
Yep! I find it is a good point!

Reply:
update within https://github.com/pagopa/io-app/pull/1936/commits/0dd7a0f466610d2a67f67ac271b98786b0bef05d